services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -391,4 +408,51 @@ public Object getUserInfo(UserAuthorization userAuthorization, String oauthClien
 			return new JWTWrapper(jwtIdToken);
 		}
 	}
+	
+	// If the client is not verified the message includes the AC team contact email iff the user id is the same as the client creator
+	protected void validateClientVerificationStatus(String clientId, String userId) throws NotFoundException {
+		validateClientVerificationStatus(clientId, () -> 
+			oauthClientDao.getOAuthClientCreator(clientId).equals(userId)
+		);
+	}
+	
+	// If the client is not verified the message includes the AC team contact email
+	protected void validateClientVerificationStatus(String clientId) throws NotFoundException {
+		validateClientVerificationStatus(clientId, () -> true);
+	}
+	
+	/**
+	 * Validates that the verified flag is true for the client with the given id, the given condition is evaluated iff the
+	 * client is not verified. If the condition evaluates to true the error message will include the email of the AC Team to
+	 * contact for verification.
+	 * 
+	 * @param clientId The id of the client to validate
+	 * @param includeACTMessageCondition A condition that is evaluated if the client is not verified to include in the error
+	 *                                   message the contact email of the AC Team.
+	 * @throws OAuthClientNotVerifiedException If the client is not verified
+	 * @throws NotFoundException               If a client with the given id does not exist
+	 */
+	private void validateClientVerificationStatus(String clientId, BooleanSupplier includeACTMessageCondition) throws NotFoundException {
+		
+		if (oauthClientDao.isOauthClientVerified(clientId)) {
+			return;
+		}
+		
+		StringBuilder errorMessage = new StringBuilder("The client is not verified yet.");
+
+		if (includeACTMessageCondition.getAsBoolean()) {
+			String actEmail = getVerificationContactEmail();
+			if (actEmail != null) {
+				errorMessage.append(" Pleast contact the Synapse Access and Compliance Team (" + actEmail + ") to verify your client.");
```

Comment:
Maybe just point to https://docs.synapse.org/articles/using_synapse_as_an_oauth_server.html instead of including the ACT email.